Index: lockfile/__init__.py
===================================================================
--- lockfile/__init__.py	(revision 78)
+++ lockfile/__init__.py	(working copy)
@@ -24,16 +24,14 @@
 >>> lock = FileLock('somefile')
 >>> print lock.is_locked()
 False
->>> with lock:
-...    print lock.is_locked()
-True
->>> print lock.is_locked()
-False
 
 >>> lock = FileLock('somefile')
 >>> # It is okay to lock twice from the same thread...
->>> with lock:
-...     lock.acquire()
+>>> lock.acquire()
+>>> try:
+...   lock.acquire()
+... finally:
+...   lock.release()
 ...
 >>> # Though no counter is kept, so you can't unlock multiple times...
 >>> print lock.is_locked()
@@ -166,7 +164,7 @@
         self.hostname = socket.gethostname()
         self.pid = os.getpid()
         if threaded:
-            self.tname = "%x-" % (threading.current_thread().ident &
+            self.tname = "%x-" % (id(threading.current_thread()) &
                                   0xffffffff)
         else:
             self.tname = ""
